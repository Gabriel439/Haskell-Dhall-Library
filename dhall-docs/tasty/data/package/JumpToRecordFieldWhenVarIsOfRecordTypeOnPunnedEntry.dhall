{-|
On this example, `dhall-docs` will detect that the variable `a` is bounded
to a record literal. It will also be aware that that the attribute `x`. When
the user clicks the `x` from `a.x`, it will jump to the `x` punned-entry
-}
let x = 1
let a = { x }

in a.x
