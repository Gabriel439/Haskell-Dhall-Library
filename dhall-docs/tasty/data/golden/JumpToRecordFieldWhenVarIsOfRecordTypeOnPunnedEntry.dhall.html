<!DOCTYPE HTML>
<html
  ><head
    ><title
    >/JumpToRecordFieldWhenVarIsOfRecordTypeOnPunnedEntry.dhall</title
    ><link href="index.css" type="text/css" rel="stylesheet"
    /><link href="https://fonts.googleapis.com/css2?family=Fira+Code:wght@400;500;600;700&amp;family=Lato:ital,wght@0,400;0,700;1,400&amp;display=swap"
            type="text/css" rel="stylesheet"
    /><script src="index.js" type="text/javascript"
    /><meta charset="UTF-8"/></head
  ><body
    ><div class="nav-bar"
      ><img src="dhall-icon.svg" class="dhall-icon"
      /><p class="package-title"
      >test-package</p
      ><div class="nav-bar-content-divider"
      /><a id="switch-light-dark-mode" class="nav-option"
      >Switch Light/Dark Mode</a></div
    ><div class="main-container"
      ><h2 class="doc-title"
        ><span class="crumb-divider"
        >/</span
        ><a href="index.html"
        >test-package</a
        ><span class="crumb-divider"
        >/</span
        ><span href="index.html" class="title-crumb"
        >JumpToRecordFieldWhenVarIsOfRecordTypeOnPunnedEntry.dhall</span></h2
      ><a data-path="/JumpToRecordFieldWhenVarIsOfRecordTypeOnPunnedEntry.dhall"
          class="copy-to-clipboard"
        ><i
          ><small
          >Copy path to clipboard</small></i></a
      ><br
      /><div class="doc-contents"
        ><p
          >On this example, <code
          >dhall-docs</code
          > will detect that the variable <code
          >a</code
          > is bounded
to a record literal. It will also be aware that that the attribute <code
          >x</code
          >. When
the user clicks the <code
          >x</code
          > from <code
          >a.x</code
          >, it will jump to the <code
          >x</code> punned-entry</p>
</div
      ><h3
      >Source</h3
      ><div class="source-code"
        ><pre
          >{-|<br
          />On this example, `dhall-docs` will detect that the variable `a` is bounded<br
          />to a record literal. It will also be aware that that the attribute `x`. When<br
          />the user clicks the `x` from `a.x`, it will jump to the `x` punned-entry<br
          />-}<br
          />let x = 1<br
          />let <span id="var7-5" data-variable="var7-5"
                      class="variable-decl"
          >a</span
          > = { <span id="var7-11" data-variable="var7-11"
                      class="variable-decl"
          >x</span
          > }<br
          /><br
          />in <a href="#var7-5" data-variable="var7-5" class="variable-use"
          >a</a
          >.<a href="#var7-11" data-variable="var7-11" class="variable-use"
          >x</a
          ><br/></pre></div></div></body></html>